src/main/java/executor/service/service/impl/ProxySourcesClientImpl.java

```diff
@@ -1,11 +1,88 @@
 package executor.service.service.impl;
 
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import executor.service.model.ProxyConfigHolder;
+import executor.service.model.ProxyCredentials;
+import executor.service.model.ProxyNetworkConfig;
 import executor.service.service.ProxySourcesClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Flux;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import static executor.service.config.properties.PropertiesConstants.PROXY_CREDENTIALS;
+import static executor.service.config.properties.PropertiesConstants.PROXY_NETWORK_CONFIG;
 
 public class ProxySourcesClientImpl implements ProxySourcesClient {
+
+    private static final Logger log = LoggerFactory.getLogger(ProxySourcesClientImpl.class);
+    public static final int DELAY = 1;
+
+    public ProxySourcesClientImpl() {
+    }
+
     @Override
-    public ProxyConfigHolder getProxy() {
-        return null;
+    public Flux<ProxyConfigHolder> getProxies() {
+        List<ProxyConfigHolder> proxyConfigHoldersPrototype = getProxyConfigHoldersPrototype();
+        validateProxies(proxyConfigHoldersPrototype);
+        return Flux.fromIterable(proxyConfigHoldersPrototype)
+                .log()
+                .delayElements(Duration.ofSeconds(DELAY))
+                .repeat();
+    }
+
+    private void validateProxies(List<ProxyConfigHolder> proxies) {
+        if(proxies == null || proxies.isEmpty()) {
+            log.error("Bad proxies list");
+            throw new IllegalArgumentException("List cannot be null or empty");
+        }
+    }
+
+    private List<ProxyConfigHolder> getProxyConfigHoldersPrototype() {
+        List<ProxyCredentials> proxyCredentials = readProxyCredentialsFromJson();
+        List<ProxyNetworkConfig> proxyNetworkConfigs = readProxyNetworkConfigFromJson();
```

Comment:
what's reason to fetch proxy host and proxy credentials separately?

Reply:
I refactored code.